### PR TITLE
Add functionSegments to parameter list of cast function in MySQL visitor

### PR DESCRIPTION
**Pull request by Afsalmc**
Changes proposed in this pull request:
  -----------------------------------------------------------------------

> Append functionSegments to parameters list in visitCast of MySQL visitor.

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/MySQLStatementVisitor.java [hunk 1]
```diff
@@ -1034,8 +1034,7 @@ public final ASTNode visitCastFunction(final CastFunctionContext ctx) {
                 result.getParameters().add((ColumnSegment) expr);
             } else if (expr instanceof LiteralExpressionSegment) {
                 result.getParameters().add((LiteralExpressionSegment) expr);
-            }
-            else if (expr instanceof FunctionSegment) {
+            } else if (expr instanceof FunctionSegment) {
```
Comment (strongduanmu) [inline parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/MySQLStatementVisitor.java:1037]: Hi @Afsalmc, does this change has sql parser test case cover?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/MySQLStatementVisitor.java (hunk 1 above)
Comment (Afsalmc) [reply parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/MySQLStatementVisitor.java:1037]: @strongduanmu No. I haven't made any changes to the test cases.

Comment (strongduanmu) [reply parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/MySQLStatementVisitor.java:1037]: What I mean is that after modifying the logic of visit, it needs to add the corresponding case to test it. Additionally, something went wrong with the CI, which also needed to be solved.

**Comment by strongduanmu** [comment]
Since this pr has long time no response, I will close it. You are welcome to submit a new PR and modify the comment issue.

**Pull request closed without merging**